workflows/ampliseq.nf

```diff
@@ -44,12 +44,24 @@ if (params.classifier) {
 
 if (params.dada_ref_taxonomy && !params.skip_taxonomy) {
     ch_dada_ref_taxonomy = Channel.fromList(params.dada_ref_databases[params.dada_ref_taxonomy]["file"]).map { file(it) }
+    if (params.addsh) {
+        if (params.dada_ref_databases[params.dada_ref_taxonomy]["shfile"]) {
+            ch_shinfo = Channel.fromList(params.dada_ref_databases[params.dada_ref_taxonomy]["shfile"]).map { file(it) }
+        } else {
+            exit 1, "SH information is not available for current selection of reference database"
+        }
+    }
 } else { ch_dada_ref_taxonomy = Channel.empty() }
 
 if (params.qiime_ref_taxonomy && !params.skip_taxonomy && !params.classifier) {
     ch_qiime_ref_taxonomy = Channel.fromList(params.qiime_ref_databases[params.qiime_ref_taxonomy]["file"]).map { file(it) }
 } else { ch_qiime_ref_taxonomy = Channel.empty() }
 
+if (params.addsh && params.cut_its == "none") {
+    log.warn "Adding SH assignments is only feasible for ITS sequences. Please use option `--cut_its` to find ITS regions in the ASV sequences, unless the given sequences are already cut to the ITS region.\n"
+}
```

Comment:
That could be (but imho also fine if not) moved to [lib/WorkflowAmpliseq.groovy](https://github.com/nf-core/ampliseq/blob/master/lib/WorkflowAmpliseq.groovy).

Reply:
Agreed that it fits better in lib/WorkflowAmpliseq.groovy